packages/cli/commands/sandbox/create.js

```diff
@@ -11,9 +12,90 @@ const { createSandbox } = require('@hubspot/cli-lib/sandboxes');
 const { loadAndValidateOptions } = require('../../lib/validation');
 const { createSandboxPrompt } = require('../../lib/prompts/sandboxesPrompt');
 const { i18n } = require('@hubspot/cli-lib/lib/lang');
+const { logErrorInstance } = require('@hubspot/cli-lib/errorHandlers');
+const {
+  ENVIRONMENTS,
+  PERSONAL_ACCESS_KEY_AUTH_METHOD,
+  DEFAULT_HUBSPOT_CONFIG_YAML_FILE_NAME,
+} = require('@hubspot/cli-lib/lib/constants');
+const {
+  personalAccessKeyPrompt,
+} = require('../../lib/prompts/personalAccessKeyPrompt');
+const {
+  updateConfigWithPersonalAccessKey,
+} = require('@hubspot/cli-lib/personalAccessKey');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { writeConfig, updateAccountConfig } = require('@hubspot/cli-lib');
+const { promptUser } = require('../../lib/prompts/promptUtils');
+const { accountNameExistsInConfig } = require('@hubspot/cli-lib/lib/config');
+const { STRING_WITH_NO_SPACES_REGEX } = require('../../lib/regex');
+const { getHubSpotWebsiteOrigin } = require('@hubspot/cli-lib/lib/urls');
 
 const i18nKey = 'cli.commands.sandbox.subcommands.create';
 
+const promptForAccountNameIfNotSet = async (updatedConfig, name) => {
+  if (!updatedConfig.name) {
+    let promptAnswer;
+    let validName = null;
+    while (!validName) {
+      promptAnswer = await promptUser([
+        {
+          name: 'name',
+          message: i18n(`${i18nKey}.enterAccountName`),
+          validate(val) {
+            if (typeof val !== 'string') {
+              return i18n(`${i18nKey}.errors.invalidName`);
+            } else if (!val.length) {
+              return i18n(`${i18nKey}.errors.nameRequired`);
+            } else if (!STRING_WITH_NO_SPACES_REGEX.test(val)) {
+              return i18n(`${i18nKey}.errors.spacesInName`);
+            }
+            return true;
+          },
+          default: name,
+        },
+      ]);
+
+      if (!accountNameExistsInConfig(promptAnswer.name)) {
+        validName = promptAnswer.name;
+      } else {
+        logger.log(
+          i18n(`${i18nKey}.errors.accountNameExists`, {
+            name: promptAnswer.name,
+          })
+        );
+      }
+    }
+    return validName;
+  }
+};
+
+const personalAccessKeyFlow = async (env, accountId, name) => {
```

Comment:
Don't worry about handling this in this PR, but I'm noticing that we're accumulating some repetitive auth-related prompts (I have some WIP work that's also adding more of it). I may try to consolidate some of it later on.